interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp

```diff
@@ -433,29 +507,99 @@ namespace cling {
+            return &(*result);
+          if (auto result = m_Libraries.GetRegisteredLib(LibPath))
+            return &(*result);
+
+          const LibraryPath* NewLibPath;
           if (searchSystemLibraries)
-            m_SysLibraries.RegisterLib(LibPath);
+            NewLibPath = m_SysLibraries.RegisterLib(LibPath);
           else
-            m_Libraries.RegisterLib(LibPath);
+            NewLibPath = m_Libraries.RegisterLib(LibPath);
+
+          // Handle lib dependencies
+          std::string RPath;
+          std::string RunPath;
+          std::vector<StringRef> Deps;
+          llvm::StringRef NewLibP = NewLibPath->GetFullName();
+          auto ObjFileOrErr =
+              llvm::object::ObjectFile::createObjectFile(NewLibP);
+          if (llvm::Error Err = ObjFileOrErr.takeError()) {
+            if (DEBUG > 1) {
+              std::string Message;
+              handleAllErrors(std::move(Err), [&](llvm::ErrorInfoBase& EIB) {
+                Message += EIB.message() + "; ";
+              });
+              cling::errs()
+                  << "Dyld::ScanForLibraries: Failed to read object file "
+                  << NewLibP << " Errors: " << Message << "\n";
+            }
+            return nullptr;
+          }
+          llvm::object::ObjectFile* BinObjF = ObjFileOrErr.get().getBinary();
+          if (BinObjF->isELF()) {
+            if (const auto* ELF = dyn_cast<ELF32LEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF32BEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF64LEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF64BEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+
+          } else if (BinObjF->isMachO()) {
+            MachOObjectFile* Obj = (MachOObjectFile*)BinObjF;
+            for (const auto& Command : Obj->load_commands()) {
+              if (Command.C.cmd == MachO::LC_LOAD_DYLIB) {
+                // Command.C.cmd == MachO::LC_ID_DYLIB ||
+                // Command.C.cmd == MachO::LC_LOAD_WEAK_DYLIB ||
+                // Command.C.cmd == MachO::LC_REEXPORT_DYLIB ||
+                // Command.C.cmd == MachO::LC_LAZY_LOAD_DYLIB ||
+                // Command.C.cmd == MachO::LC_LOAD_UPWARD_DYLIB ||
+                MachO::dylib_command dylibCmd =
+                    Obj->getDylibIDLoadCommand(Command);
+                Deps.push_back(StringRef(Command.Ptr + dylibCmd.dylib.name));
+              } else if (Command.C.cmd == MachO::LC_RPATH) {
+                MachO::rpath_command rpathCmd = Obj->getRpathCommand(Command);
+                CombinePaths(RPath, Command.Ptr + rpathCmd.path);
+              }
+            }
```

Comment:
What about COFF? `DLL`s also have that info.

Reply:
Can you point us to where you found that information? We do not seem to have access to `rpath` & co in the `COFFObjectFile`.

Reply:
With "that info" I meant the library dependencies which you collect for ELF as `DT_NEEDED`. IIUC an example for COFF is [here](https://github.com/llvm/llvm-project/blob/1128311a19179ceca799ff0fbc4dd206ab56e560/llvm/tools/llvm-objdump/COFFDump.cpp#L350)

Reply:
This pull request fixes a particular issue which in essence is a deployment technique on the unix platforms. I am hesitant adding such logic for COFF as I am not sure if that is relevant to that problem but also I am not sure what is the behavior of their dynamic linker. I'd pursue the approach from the past -- we will add the windows support if we need it in future.

Reply:
We *will* need the `DT_NEEDED` replacement for COFF to make ACLiC work when loading ACLiC libraries, without searching for libraries that can resolve unresolved symbols. I'm okay if this is addressed in a subsequent PR, but without that we cannot progress with https://github.com/root-project/root/issues/7774 - https://github.com/root-project/root/pull/5235 was excluding this case knowing that `DT_NEEDED` support would come.

Reply:
Let's do that, as we agreed, in another PR if required.